spectrum.go

```diff
@@ -0,0 +1,185 @@
+package cloudflare
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// SpectrumApplication defines a single Spectrum Application.
+type SpectrumApplication struct {
+	ID            string                        `json:"id,omitempty"`
+	Protocol      string                        `json:"protocol,omitempty"`
+	IPv4          bool                          `json:"ipv4,omitempty"`
+	DNS           SpectrumApplicationDNS        `json:"dns,omitempty"`
+	OriginDirect  []string                      `json:"origin_direct,omitempty"`
+	OriginPort    int                           `json:"origin_port,omitempty"`
+	OriginDNS     *SpectrumApplicationOriginDNS `json:"origin_dns,omitempty"`
+	IPFirewall    bool                          `json:"ip_firewall,omitempty"`
+	ProxyProtocol bool                          `json:"proxy_protocol,omitempty"`
+	TLS           string                        `json:"tls,omitempty"`
+	CreatedOn     *time.Time                    `json:"created_on,omitempty"`
+	ModifiedOn    *time.Time                    `json:"modified_on,omitempty"`
+}
+
+// SpectrumApplicationDNS holds the external DNS configuration for a Spectrum
+// Application.
+type SpectrumApplicationDNS struct {
+	Type string `json:"type"`
+	Name string `json:"name"`
+}
+
+// SpectrumApplicationOriginDNS holds the origin DNS configuration for a Spectrum
+// Application.
+type SpectrumApplicationOriginDNS struct {
+	Type  string   `json:"type"`
+	Names []string `json:"names"`
+	TTL   int      `json:"ttl"`
+}
+
+// SpectrumApplicationDetailResponse is the structure of the detailed response
+// from the API.
+type SpectrumApplicationDetailResponse struct {
+	Success  bool                `json:"success"`
+	Errors   []string            `json:"errors"`
+	Messages []string            `json:"messages"`
+	Result   SpectrumApplication `json:"result"`
+}
+
+// SpectrumApplicationsDetailResponse is the structure of the detailed response
+// from the API.
+type SpectrumApplicationsDetailResponse struct {
+	Success  bool                  `json:"success"`
+	Errors   []string              `json:"errors"`
+	Messages []string              `json:"messages"`
+	Result   []SpectrumApplication `json:"result"`
+}
+
+// DeletedSpectrumApplicationResponse defines the API response when deleting a
+// Spectrum application.
+type DeletedSpectrumApplicationResponse struct {
```

Comment:
Not sure we actually need a specific struct type here, just to catch the ID we know already. Other delete methods usually don't return anything.

What's your opinion?

Reply:
This was only to support the `Delete` method which is needed. I'll get rid of this.